libs/feature/lesson/src/lib/lesson-editor.tsx

```diff
@@ -0,0 +1,120 @@
+import { DialogActions, OnDialogCloseFn, showToast, Tab, Tabs } from "@self-learning/ui/common";
+import { LessonFormModel } from "@self-learning/teaching";
+import { useRequiredSession } from "@self-learning/ui/layouts";
+import { useState } from "react";
+import { FormProvider, useForm } from "react-hook-form";
+import { createEmptyLesson, lessonSchema } from "@self-learning/types";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { LessonContentEditor } from "../../../teaching/src/lib/lesson/forms/lesson-content";
+import { QuizEditor } from "../../../teaching/src/lib/lesson/forms/quiz-editor";
+import { LessonInfoEditor } from "../../../teaching/src/lib/lesson/forms/lesson-info";
+
+export async function onLessonCreatorClosed(
+	onClose: () => void,
+	createLessonAsync: (lesson: LessonFormModel) => any,
+	lesson?: LessonFormModel
+) {
+	if (!lesson) {
+		onClose();
+		return;
+	}
+	try {
+		console.log("Creating lesson...", lesson);
+		const result = await createLessonAsync(lesson);
+		showToast({ type: "success", title: "Lernheit erstellt", subtitle: result.title });
+		onClose();
+	} catch (error) {
+		console.error(error);
+		showToast({
+			type: "error",
+			title: "Fehler",
+			subtitle: "Lerneinheit konnte nicht erstellt werden."
+		});
+	}
+}
+
+export async function onLessonEditorClosed(
```

Comment:
die kopplung mit der EditLessonPage Komponente ist recht hoch. Separieren in eine andere datei wirkt nicht auf den ersten blick nicht sinnig.